builder/optimizations.cc

```diff
@@ -252,11 +255,12 @@ class buffer_aliaser : public node_mutator {
   symbol_map<buffer_info> buffers;
 
   static bool alias_compatible(
```

Comment:
This function looks like a const, but isn't anymore which might be confusing, so I think it would be good to add a comment about what it does.